src/lua_openssl.cc

```diff
@@ -0,0 +1,562 @@
+extern "C" {
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <lua5.2/lua.h>
+#include <lua5.2/lualib.h>
+#include <lua5.2/lauxlib.h>
+
+#include <openssl/err.h>
+#include <openssl/dh.h>
+#include <openssl/ssl.h>
+#include <openssl/conf.h>
+#include <openssl/engine.h>
+}
+
+// SSL debug
+#define SSL_WHERE_INFO(ssl, w, flag, msg) {                \
+    if(w & flag) {                                         \
+      printf("+ %s: ", name);                              \
+      printf("%20.20s", msg);                              \
+      printf(" - %40.40s ", SSL_state_string_long(ssl));   \
+      printf(" - %5.10s ", SSL_state_string(ssl));         \
+      printf("\n");                                        \
+    }                                                      \
+  }
+
+typedef void(*info_callback)(const SSL*, int, int);
+
+namespace {
+	// C/C++ functions
+	// debug
+	void ssl_info_callback(const SSL* ssl, int where, int ret, const char* name) {
+
+		if(ret == 0) {
+			printf("-- ssl_info_callback: error occured.\n");
+			return;
+		}
+
+		SSL_WHERE_INFO(ssl, where, SSL_CB_LOOP, "LOOP");
+		SSL_WHERE_INFO(ssl, where, SSL_CB_HANDSHAKE_START, "HANDSHAKE START");
+		SSL_WHERE_INFO(ssl, where, SSL_CB_HANDSHAKE_DONE, "HANDSHAKE DONE");
+	}
+
+	void ssl_server_info_callback(const SSL* ssl, int where, int ret) {
+		ssl_info_callback(ssl, where, ret, "server");
+	}
+
+	void ssl_client_info_callback(const SSL* ssl, int where, int ret) {
```

Comment:
@LuxoftAKutsan We need these proxies to be able to print role of ATF in secure connection (client or server). Function `ssl_info_callback()` can not be used as callback directly because it has inappropriate function signature:
callback function signature: `callback( SSL *ssl, int where, int ret)`
ssl_info_callback signature:  `ssl_info_callback(const SSL* ssl, int where, int ret, const char* name)`